### DISCUSSION TITLE
UR10 Flange Plane Flipped?

### DISCUSSION THREAD

**Discussion by architektpetr** [visose/Robots#118] [Q&A] [answered]
Hi. I was having an issue with Bartlett-UR10 setup. The tool is oriented correctly in the simulation (based on calibration data), the real robot position is correct (the connector pin in both the simulation and reality is at the same position), but the tool is 90° rotated. I noticed that flange's plane (taken from P output of Simulation component) is not oriented correctly based on the tool documentation. For now I will try to figure out some workaround (probably modify the XML file?), but it's very confusing behavior. 

EDIT: I also noticed, that the input TCP point X and Y are switched in the script export. Not in the script itself. Don't know why...

**Comment by visose** [comment, marked as answer]
Hi, to orient the tool in grasshopper, follow the instructions under the wiki here, not the UR documentation. This plugin follows a unified way of setting up the tool orientation that is the same for all robot models, and then converts it depending on the robot type.

If the simulation and real robot look identical, including flange, but the real tool is rotated 90 degrees, it means you accidentally attached the tool 90 different from the virtual tool.
